org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IBuildSupport.java

```diff
@@ -70,13 +72,24 @@ default void refresh(IResource resource, CHANGE_TYPE changeType, IProgressMonito
 			return;
 		}
 		if (changeType == CHANGE_TYPE.DELETED) {
+			if (IJavaProject.CLASSPATH_FILE_NAME.equals(resource.getName())) {
```

Comment:
Need to check if CLASSPATH_FILE_NAME is at the root of its project (else it might be harmless). Then, it
should send a message to the user saying ".classpath is required for providing proper Java support. Clean the server workspace to recreate it". (or re-open the client, if that import on re-open works -it doesn't at the moment)